src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -276,7 +276,7 @@ public static TracerSettings FromDefaultSources()
         /// AppSettings where available, and a local datadog.json file, if present.
         /// </summary>
         /// <returns>A new <see cref="IConfigurationSource"/> instance.</returns>
-        public static CompositeConfigurationSource CreateDefaultConfigurationSource()
+        public static IConfigurationSource CreateDefaultConfigurationSource()
         {
             return GlobalSettings.CreateDefaultConfigurationSource();
```

Comment:
I think the "right" thing to do is to leave `GlobalSettings.CreateDefaultConfigurationSource()` unchanged since it just adds various sources, and then here in `TracerSettings.CreateDefaultConfigurationSource()` we could wrap the returned object with the fallback source. What do you think of that idea?